src/app/src/main/kotlin/org/imperial/mrc/hint/logging/LogElapsedTimeCallback.kt

```diff
@@ -0,0 +1,17 @@
+package org.imperial.mrc.hint.logging
+
+import java.time.Duration
+import java.time.Instant
+
+inline fun <T> logDurationOf(
+    callback: () -> T,
+    logger: GenericLogger,
+): T
+{
+    val start = Instant.now()
+    val response = callback()
+    val end = Instant.now()
+    val timeElapsed = Duration.between(start, end).toMillis()
+    logger.info("ADR request time elapsed: $timeElapsed")
+    return response
+}
```

Comment:
Strange that codecov thinks this isn't tested, when it clearly is, in `LogElapsedTimeCallbackTests`!